app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -230,18 +230,20 @@ def parse_resource_group(resource_group)
         return uid, new_result
       end
 
-      def parse_series(s)
-        name = uid = s.name.downcase
+      def parse_series(series)
+        uid = series.name.downcase
+
         new_result = {
-          :type           => "ManageIQ::Providers::Azure::CloudManager::Flavor",
+          :type           => ManageIQ::Providers::Azure::CloudManager::Flavor.name,
           :ems_ref        => uid,
-          :name           => name,
-          :cpus           => s.number_of_cores, # where are the virtual CPUs??
-          :cpu_cores      => s.number_of_cores,
-          :memory         => s.memory_in_mb.megabytes,
-          :root_disk_size => s.os_disk_size_in_mb.megabytes,
-          :swap_disk_size => s.resource_disk_size_in_mb.megabytes
+          :name           => series.name.split('_').map(&:capitalize).join('_'),
```

Comment:
Why are you not saving the name as it is pulled in from Azure? I would think the name we send back  **_to_** Azure during provisioning should match what we receive _**from**_ Azure in refresh.

Reply:
I thought it might be best to make sure, but I can undo that, as I guess it could backfire.